api/types/filters/errors.go

```diff
@@ -0,0 +1,24 @@
+package filters
+
+import "fmt"
+
+// invalidFilter indicates that the provided filter or its value is invalid
+type invalidFilter struct {
+	Filter string
+	Value  []string
+}
+
+func (e invalidFilter) Error() string {
+	msg := "invalid filter"
+	if e.Filter != "" {
+		msg += " '" + e.Filter
+		if e.Value != nil {
+			msg += fmt.Sprintf("=%s", e.Value)
```

Comment:
Ah, good one, thanks!

Reply:
🤦 doh! I hate the filter implementation.

Reply:
Hmm, I'm thinking that this could be changed to `msg = fmt.Sprintf("%s=%s", msg, e.Value)` - if we already do the Sprintf, let it handle the concatenation too.

Reply:
I think it was done because the value is "optionally there". Current code seems fine from that perspective.